AsterNET.ARI/ARIClient.cs

```diff
@@ -136,6 +136,9 @@ private void _eventProducer_OnConnectionStateChanged(object sender, EventArgs e)
         {
             if (_eventProducer.State != ConnectionState.Open)
                 Reconnect();
+
+            if (OnConnectionStateChanged != null)
```

Comment:
I am happy to go with removing it. As I said, it was a relic more from before Middleware was added. just amend your fork and push the changes, the pull request should update itself.